### PR TITLE
SVM: Handle multiple total data products (TDP) when first TDP processed has no sources

### PR DESCRIPTION
**Pull request by mdlpstsci**
Ensure the segmentation image is not None upon return from detect_and_eval_segments()and handle the situation appropriately though additional logic.  Added informational log messages regarding lack of sources found and catalog status. Protect against divide by zero when computing the ratio of "Gaussian big islands" vs "RW big islands".

### CODE REVIEW

File: drizzlepac/haputils/catalog_utils.py
Comment (stsci-hack) [inline drizzlepac/haputils/catalog_utils.py:1481]: If 'rw_bs' == 0, the variable 'ratio_cg2rw_bigsource' never gets initialized with a value, yet it gets used later (line 1586).  On the face of it, this would cause an exception unless having 'rw_bs' == 0 somehow prevents the logic on line 1586 from ever being called.  Please clarify to insure no Exceptions get thrown.

**Review by stsci-hack — CHANGES_REQUESTED** [review]
Overall looks fine.  Just one bit of logic which needs to be clarified before approval.

File: drizzlepac/haputils/catalog_utils.py
Comment (mdlpstsci) [reply drizzlepac/haputils/catalog_utils.py:1482]: Oops.  Fixed in 9a8b0d5.

File: drizzlepac/haputils/catalog_utils.py
Comment (stsci-hack) [inline drizzlepac/haputils/catalog_utils.py:1480]: Should this be 'ratio_cg2rw_bigsource' instead?

**Review by stsci-hack — COMMENTED** [review]
It is not clear that the variable was initialized correctly... please confirm.

File: drizzlepac/haputils/catalog_utils.py
Comment (stsci-hack) [inline drizzlepac/haputils/catalog_utils.py:1751]: The logic here for 'seg_img is None' duplicates the logic at the start of the method '_evaluate_segmentation_image'.  Removing this entire if...else block and adding these log messages to the method's own 'if seg_img is None' block of code would simplify the code by putting this check in only one place.

**Review by stsci-hack — APPROVED** [review]
My comment doesn't require another revision, but is something to consider for later.  Otherwise, I think this PR is ready to merge.

File: drizzlepac/haputils/catalog_utils.py
Comment (mdlpstsci) [reply drizzlepac/haputils/catalog_utils.py]: Fixed in ada6517.

File: drizzlepac/haputils/catalog_utils.py
Comment (mdlpstsci) [reply drizzlepac/haputils/catalog_utils.py:1751]: Does not hurt anything.

**Pull request merged**
